ERCS/erc-7524.md

```diff
@@ -0,0 +1,211 @@
+    - zk proof of solvency — if you want two exchanges to prove they know a set of private keys that hold some balance, you need a way to ensure that two exchanges aren’t both claiming the same address, while keeping it private
+    
+As such, a deterministic value based on the Ethereum account’s ECDSA keypair is a necessary component of ensuring one action per user and enables all these applications on Ethereum.
+    
+
+## Specification
+
+We propose a new signature standard that offers the following properties, to be implemented for standard ECDSA keys within wallets:
+
+1. It produces signatures that contain a deterministic component and a nondeterministic component. The deterministic component may be used as a *nullifier*.
+2. Signers can use existing secpk256k1 keypairs, such as those in hardware wallets that support Ethereum accounts. As a consequence, secret keys can remain in secure enclaves if there is a generator point multiplication API into the enclave (which Ledger for instance has).
+
+### Parameters
+
+This scheme uses the secp256k1 curve, defined in SEC 2: Recommended Elliptic Curve Domain Parameters v2, page 9.
+
+We use the following notation to refer to the parameters of this curve:
+
+- $g$: the base point (also called the generator) of the curve.
+- $p$: the order of the curve.
+- $F_p$: the finite field whose order is $p$.
+
+Note we use exponential notation to denote elliptic curve scalar multiplications.
+
+### Public key encoding functions
+
+### SEC1
+
+This scheme uses the SEC1 elliptic curve point encoding scheme defined in SEC 1: Elliptic Curve Cryptography v2 . Point compression is used. We use the notation $\mathsf{sec1}(pk)$ to denote the compressed encoding of secp256k1 curve point $pk$ as a bytestring of length 33.
+
+### Hash functions
+
+**SHA256**
+
+This scheme uses the SHA256 hash function defined in IETF RFC4634.
+
+In this document, we use the notation $\mathsf{sha256}(a_1,.. a_n)$ to denote the sha256 digest of the concatentation of $n$ values $a_1, ..., a_n$. The digest should then be interpreted as a big-endian value in the secp256k1 scalar field.
+
+### Hash-to-curve
+
+We use the notation $\mathsf{htc}([a_1, ..., a_n])$ to denote the elliptic curve point which is the result of the `secp256k1_XMD:SHA-256_SSWU_RO_ (IETF Draft draft-irtf-cfrg-hash-to-curve-16 Appendix J.8.1)` hash-to-curve algorithm over the concatentation of $n$ values $a_1, ..., a_n$.
+
+### Key generation
+
+A *keypair* comprises of $(sk, pk)$, defined as such:
+
+- $sk$: The user's secret key, which is a cryptographically secure random scalar in the field $F_p$.
+- $pk$: The user's public key, defined as $g^{sk}$, which is a point on the secp256k1 curve.
+
+### Signature generation
+
+This scheme builds upon the Chaum-Pedersen signature scheme. Given a 32-byte message $m$ and a keypair $(sk, pk)$, a  user may generate a signature as such:
```

Comment:
Might be a good idea to cite the original paper. I _think_ this is the correct one, but I'm terrible at finding academic papers, so please confirm.

````suggestion
This scheme builds upon the Chaum-Pedersen signature scheme. [^1] Given a 32-byte message $m$ and a keypair $(sk, pk)$, a  user may generate a signature as such:

[^1]:
    ```csl-json
    {
      "DOI": "10.1007/3-540-48071-4_7",
      "URL": "https://link.springer.com/content/pdf/10.1007/3-540-48071-4_7.pdf",
      "publisher-place": "Berlin, Heidelberg",
      "author": [
        {
          "given": "David",
          "family": "Chaum"
        },
        {
          "given": "Torben Pryds",
          "family": "Pedersen"
        }
      ],
      "container-title": "Advances in Cryptology — CRYPTO' 92",
      "editor": [
        {
          "given": "Ernest F.",
          "family": "Brickell"
        }
      ],
      "type": "paper-conference",
      "id": "10.1007/3-540-48071-4_7",
      "citation-label": "10.1007/3-540-48071-4_7",
      "ISBN": "978-3-540-48071-6",
      "issued": {
        "date-parts": [
          [
            1993
          ]
        ]
      },
      "page": "89-105",
      "publisher": "Springer Berlin Heidelberg",
      "title": "Wallet Databases with Observers"
    }
    ```
````